hasher-matcher-actioner/hmalib/common/aws_dataclass.py

```diff
@@ -49,6 +51,13 @@ def py_to_aws(py_field: t.Any, in_type: t.Optional[t.Type[T]] = None) -> T:
 
     check_type = origin or in_type
 
+    if isinstance(check_type, t.ForwardRef):
```

Comment:
@Dcallies and I collaborated on getting the refactor in this PR to work. One thing I'd run into was that one of the fields of the base dataclass was being typed using a forward reference, which the code here doesn't support. So we decided to include a helpful error message in case someone else runs into this in the future.